File: Python Assignments/Python Assignment 2/Assignment2_5.py
#   5. Write a program which accept one number for user and check whether
#      number is prime or not.

def ChkPrime(No):
    count = 0
    for i in range(1,No+1):
        if No%i == 0:
            count = count + 1
            if count > 2:
                break
    return count
                
def main():
    Number = int(input("Enter your number : "))
    if Number < 0:
        No = -Number
        count = ChkPrime(No)
    else:
        count = ChkPrime(Number)

    if count > 2:
        print("{0} is not Prime Number".format(Number))
    elif count < 2:
        print("{0} is neither prime nor composite".format(Number))
    else:
        print("{0} is Prime Number".format(Number))

if __name__ == "__main__":
    main()

#   Output:
#   D:\Python\Marvellous Python\Python Assignment 2>python Assignment2_5.py
#   Enter your number : -1
#   -1 is neither prime nor composite

#   D:\Python\Marvellous Python\Python Assignment 2>python Assignment2_5.py
#   Enter your number : 0
#   0 is neither prime nor composite

#   D:\Python\Marvellous Python\Python Assignment 2>python Assignment2_5.py
#   Enter your number : 1
#   1 is neither prime nor composite

#   D:\Python\Marvellous Python\Python Assignment 2>python Assignment2_5.py
#   Enter your number : 5
#   5 is Prime Number

#   D:\Python\Marvellous Python\Python Assignment 2>python Assignment2_5.py
#   Enter your number : 1331
#   1331 is not Prime Number

File: Typescript_Assignments/Typescript_Assignment_2/Assignment2_5.ts
//  5. Write a typescript program which contains one function named as ChkString.
//     That function accept one string and check whether that string contains
//     "Marvellous" word or not.

function ChkString(Str: string): boolean 
{
    var Word: string = "Marvellous";
    var i: number = 0;
    var j: number = 0;
    var WordLen: number = Word.length;
    var StrLen: number = Str.length;

    while (i <= StrLen - WordLen) 
    {
        for (j = 0; j < WordLen; j++) 
        {
            if (Str[i + j] !== Word[j]) 
            {
                break; 
            }
        }

        if (j === WordLen) 
        {
            return true;
        }

        i++;
    }

    return false; 
}

console.log("Start of application\n");

if(ChkString("Pune Kothrud Marvellous Infosystems") == true)
{
    console.log("String contains Marvellous in it.");
}
else
{
    console.log("String does not contains Marvellous in it.");
}

console.log("\nEnd of application");

/*

    Output:
    D:\Marvellous_Infosystems\Angular_with_MEAN_FullStack\Angular_Assignments\Typescript_Assignmnet_02>tsc Assignment2_5.ts  

    D:\Marvellous_Infosystems\Angular_with_MEAN_FullStack\Angular_Assignments\Typescript_Assignmnet_02>node Assignment2_5.js
    Start of application

    String contains Marvellous in it.

    End of application

*/